src/main/java/domain/Invoice.kt

```diff
@@ -15,20 +15,25 @@ import javax.persistence.Id
 import javax.persistence.ManyToOne
 import javax.persistence.NamedQuery
 import javax.persistence.Table
+import javax.persistence.Temporal
+import javax.persistence.TemporalType
 
 @Entity
 @NamedQuery(name = "Invoice.allInvoices", query = "SELECT i FROM Invoice i")
 @Table(name = "invoice")
 data class Invoice(
 
     @Enumerated(EnumType.STRING)
-    var generationType: InvoiceGenerationType,
+    val generationType: InvoiceGenerationType,
     @Enumerated(EnumType.STRING)
     var state: InvoiceState = InvoiceState.OPEN,
     @ManyToOne
-    var owner: Profile? = null,
+    val profile: Profile? = null,
```

Comment:
If you default it to null, why is it in the constructor? Try making  it a `lateinit var` as it will have to be filled in before it's saved

Reply:
Fixed